includes/abstracts/llms-abstract-admin-wizard.php

```diff
@@ -0,0 +1,390 @@
+		 *
+		 * The dynamic portion of this hook, `$step`, refers to the slug of the current step.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $text Button text string.
+		 */
+		return apply_filters( "llms_setup_wizard_get_{$step}_skip_text", $this->get_steps()[ $step ]['skip'] ?? '' );
+
+	}
+
+	/**
+	 * Get the URL to a step
+	 *
+	 * @since [version]
+	 *
+	 * @param string $step Step slug.
+	 * @return string
+	 */
+	private function get_step_url( string $step ): string {
+		return add_query_arg(
+			array(
+				'page' => 'llms-' . $this->type,
+				'step' => $step,
+			),
+			admin_url()
+		);
+	}
+
+	/**
+	 * Get an array of step slugs => titles
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_steps(): array {
+
+		/**
+		 * Filter the steps included in the setup wizard
+		 *
+		 * @since [version]
+		 *
+		 * @param string[] $steps Array of setup wizard steps. The array key is the slug/id of the step and the array value
+		 *                        is the step's title displayed in the wizard's navigation.
+		 */
+		return apply_filters( 'llms_setup_wizard_steps', $this->steps );
+
+	}
+
+	/**
+	 * Output the HTML content of the setup page
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function output(): void {
+		$views_dir = trailingslashit( esc_attr( $this->views_dir ) );
+		$step_html = '';
+		$steps     = $this->get_steps();
+		$current   = $this->get_current_step() ?? 'intro';
+		$prev      = $this->get_prev_step();
+		$next      = $this->get_next_step();
+		$transient = get_transient( $this->transient_key ) ?? [];
+
+		if ( in_array( $current, array_keys( $steps ), true ) ) {
+
+			ob_start();
+			include $views_dir . 'step-' . $current . '.php';
+			$step_html = ob_get_clean();
+
+		}
+
+		/**
+		 * Filter the HTML of a step within the setup wizard.
+		 *
+		 * The dynamic portion of this hook, `$current`, refers to the slug of the current step.
+		 *
+		 * This filter can be used to output the HTML for a custom step in the setup wizard.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $step_html HTML of the step.
+		 * @param LLMS_Admin_Setup_Wizard $wizard Setup wizard class instance.
+		 */
+		$step_html = apply_filters( "llms_setup_wizard_{$current}_html", $step_html, $this );
+
+		include $views_dir . 'main.php';
+
+	}
+
+	/**
+	 * Handle saving data during setup
+	 *
+	 * @since [version]
+	 *
+	 * @return null|WP_Error
+	 */
+	public function save() {
+
+		if ( ! isset( $_POST['llms_setup_nonce'] ) || ! llms_verify_nonce( 'llms_setup_nonce', 'llms_setup_save' ) || ! current_user_can( 'manage_lifterlms' ) ) {
+			return null;
+		}
+
+		$response = new WP_Error( 'llms-setup-save-invalid', __( 'There was an error saving your data, please try again.', 'lifterlms' ) );
+
+		$step = llms_filter_input( INPUT_POST, 'llms_setup_save' );
+
+		if ( method_exists( $this, 'save_' . $step ) ) {
+			$response = call_user_func( array( $this, 'save_' . $step ) );
```

Comment:
```suggestion
			$response = $this->{"save_{$step}"}();
```
should work as well without using `call_user_func()`?